### PR TITLE
[ios] Reland "[ios] use python script to generate extension safe frameworks and code sign them" #46004"

### PR DESCRIPTION
**Pull request by cyanglaz**
Relands https://github.com/flutter/engine/pull/45781
The Flutter.framework and the sim folders are mistakenly added to the artifact, this PR removed those files along with re-landing the original changes 

https://github.com/flutter/flutter/pull/134966 should pass with this change. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: sky/tools/create_full_ios_framework.py [hunk 1]
```diff
@@ -192,12 +233,19 @@ def zip_archive(dst):
       'Flutter.xcframework',
       'entitlements.txt',
       'without_entitlements.txt',
+      'extension_safe/Flutter.xcframework',
```
Comment (cyanglaz) [inline sky/tools/create_full_ios_framework.py:236]: In the original PR, `extension_safe/Flutter.framework` and `extension/sim/` were mistakenly added to this list. Only `Flutter.xcframework` should be uploaded, just like the outside folder, see line 233 above.

Comment (christopherfujino) [reply sky/tools/create_full_ios_framework.py:236]: :+1:

**Review by christopherfujino — APPROVED** [review]
LGTM

File: sky/tools/create_full_ios_framework.py [hunk 2]
```diff
@@ -173,7 +212,9 @@ def zip_archive(dst):
   ios_file_with_entitlements = ['gen_snapshot_arm64']
   ios_file_without_entitlements = [
       'Flutter.xcframework/ios-arm64/Flutter.framework/Flutter',
-      'Flutter.xcframework/ios-arm64_x86_64-simulator/Flutter.framework/Flutter'
+      'Flutter.xcframework/ios-arm64_x86_64-simulator/Flutter.framework/Flutter',
+      'extension_safe/Flutter.xcframework/ios-arm64/Flutter.framework/Flutter',
+      'extension_safe/Flutter.xcframework/ios-arm64_x86_64-simulator/Flutter.framework/Flutter'
```
Comment (cyanglaz) [inline sky/tools/create_full_ios_framework.py:217]: These are the new artifacts that I'm adding @chingjun

**Comment by cyanglaz** [comment]
Auto engine roll will fail after this lands, it will require a manual roll in https://github.com/flutter/flutter/pull/134966

cc @jonahwilliams as the current sheriff

**Pull request merged**